pkg/validate/security_context.go

```diff
@@ -1133,3 +1151,24 @@ func checkSetHostname(rc internalapi.RuntimeService, containerID string, setable
 		Expect(err).To(HaveOccurred(), msg)
 	}
 }
+
+func checkMountAndProcessLabels(rc internalapi.RuntimeService, containerID string, privileged bool) {
+	// Check that the mount label is set for privileged containers
+	cmd := []string{"ls", "-lZ", "bin"}
+	stdout, stderr, err := rc.ExecSync(containerID, cmd, time.Duration(defaultExecSyncTimeout)*time.Second)
+	msg := fmt.Sprintf("cmd %v, stdout %q, stderr %q", cmd, stdout, stderr)
+	Expect(err).NotTo(HaveOccurred(), msg)
+	Expect(string(stdout)).To(ContainSubstring("object_r:container_file_t"))
+
+	// Check that the correct process label is set for privileged and unprivileged containers
+	cmd = []string{"cat", "/proc/self/attr/current"}
+	stdout, stderr, err = rc.ExecSync(containerID, cmd, time.Duration(defaultExecSyncTimeout)*time.Second)
+	msg = fmt.Sprintf("cmd %v, stdout %q, stderr %q", cmd, stdout, stderr)
+	Expect(err).NotTo(HaveOccurred(), msg)
+
+	if privileged {
+		Expect(string(stdout)).To(ContainSubstring("system_r:spc_t"))
```

Comment:
not sure all this is going to work on non rhel systems (like fedora).

btw, I was thinking about having 2 separate tests:

- mount label: just have a cmd for the pod that does `ls -ldZ /` and make sure level is applied even if privileged
- process label: just have a cmd that `cat /proc/self/attr/current`

all of that to avoid having pods running `top`

Reply:
For mount_label we can grep for it in /proc/1/mountinfo.

Reply:
The tests failed on Travis (based on Ubuntu):

```
Expected
        <string>: lrwxrwxrwx 1 root root ? 7 Feb  7  2018 bin -> usr/bin
    to contain substring
        <string>: object_r:container_file_t
    /home/travis/gopath/src/github.com/kubernetes-sigs/cri-tools/pkg/validate/security_context.go:1161
```

Reply:
The labels work fine on my fedora 28. Also split the it into 2 separate tests.

Reply:
ubuntu doesn't have selinux enabled/installed by default for sure

Reply:
@runcom I moved the test to selinux_linux.go, which checks if selinux is enabled before running the tests. That should be good right?

Reply:
yup yup! sorry overlooked that :+1: